File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/AcademicWarningPolicy.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class AcademicWarningPolicy : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string PolicyCode { get; set; } = string.Empty;
    public string PolicyTitleAr { get; set; } = string.Empty;
    public int WarningCategory { get; set; } // 1=AcademicGrade, 2=AbsenceCount, 3=BehavioralInfraction
    public decimal ThresholdValue { get; set; } // e.g. 50.0m for grade, 15.0m for absence
    public int ActionRequired { get; set; } // 1=ParentNotification, 2=CounselorMeeting, 3=Suspension
    public string? PolicyTitleEn { get; set; }
    public int ConsecutiveOccurrenceLimit { get; set; }
    public bool AutoTriggerNotification { get; set; } = true;
    public long? EscalationPolicyId { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation Property
    public virtual School? School { get; set; }
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/academic-warning-policy.ts
import type { RecordStatus, WarningCategory, ActionRequired } from './common';

export interface AcademicWarningPolicy {
  id: number;
  schoolId: number;
  policyCode: string;
  policyTitleAr: string;
  warningCategory: WarningCategory;
  thresholdValue: number;
  actionRequired: ActionRequired;
  policyTitleEn: string | null;
  consecutiveOccurrenceLimit: number;
  autoTriggerNotification: boolean;
  escalationPolicyId: number | null;
  isActive: boolean;
}

export type CreateAcademicWarningPolicyDto = Omit<AcademicWarningPolicy, 'id' | 'isActive'>;

export type UpdateAcademicWarningPolicyDto = Omit<AcademicWarningPolicy, 'schoolId' | 'isActive'>;
